testplan/testing/base.py

```diff
@@ -890,7 +889,9 @@ def dry_run(self):
 
         return result
 
-    def run_testcases_iter(self, testsuite_pattern="*", testcase_pattern="*"):
+    def run_testcases_iter(
```

Comment:
just wondering if we can only keep `suites_cases`

Reply:
This is a major point here. I was afraid to get rid of them for the moment as suites_cases is dynamically changing based on the shallow report entry passed down on put requests to the backend. Allowing it to be None let me be on the safe side so this is more of a defensive programming here. But let us discuss, because I think you feeling is correct and I share that only suites_cases (perhaps with some tweaking) could be enough.

Reply:
Actually, I think at this point the only real change would be to refactor our testing to mock the shallow entry in the new sense. suites_cases is never None as long as the filter matches at least a suite level entry (or something below that), but the way we run interactive mode without UI in our tests would lead to ~5-10 failures. I can think of cleaning this up as a hygiene item, but perhaps it is risky to put it at that priority.